buildconfig/Jenkins/Conda/cppcheck.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash -ex
+
+SCRIPT_DIR=$(dirname "$0")
+# If errors slip through to master this can be used to set a non-zero
+# allowed count while those errors are dealt with. This avoids breaking all
+# builds for all developers
+ALLOWED_ERRORS_COUNT=0
+
+if [[ ${JOB_NAME} == *pull_requests* ]]; then
```

Comment:
Can we move this `if` block to the calling script and this script just do the build?
It would also be good to pass the `JOB_NAME` in as a parameter when the script is called so that the Jenkins parameters are not hard-coded in the scripts.